hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/CassAlertsServiceImpl.java

```diff
@@ -488,34 +565,30 @@ private boolean filterByStatuses(String tenantId, Set<String> alertsId, AlertsCr
         }
 
         if (statuses.size() > 0) {
-            filterByStatus = true;
             PreparedStatement selectAlertStatusByTenantAndStatus = CassStatement.get(session,
-                    CassStatement.SELECT_ALERT_STATUS_BY_TENANT_AND_STATUS);
+                    CassStatement.SELECT_ALERT_STATUS);
             List<ResultSetFuture> futures = statuses.stream().map(status ->
                     session.executeAsync(selectAlertStatusByTenantAndStatus.bind(tenantId, status.name())))
                     .collect(Collectors.toList());
-
             List<ResultSet> rsAlertStatuses = Futures.allAsList(futures).get();
+
+            Set<String> alertIds = new HashSet<>();
             rsAlertStatuses.stream().forEach(r -> {
                 for (Row row : r) {
                     String alertId = row.getString("alertId");
-                    alertsId.add(alertId);
+                    alertIds.add(alertId);
                 }
             });
-            /*
-                If there is not alertId but we have triggersId means that we have an empty result.
-                So we need to sure a alertId to mark that we have an empty result for future joins.
-             */
-            if (alertsId.isEmpty()) {
-                alertsId.add("no-result-fake-alert-id");
```

Comment:
I couldn't figure out what this was for, despite the comment.  I need to know better what this is before adding anything back because it looks "hacky".